### PR TITLE
feat: Add `entities_to_dict()` and `entities_to_bigquery()` to `Document` wrapper

### PR DESCRIPTION
**Pull request by holtskinner**
- Uploads entities to an existing dataset, creates new table if it doesn't already exist.

## Example Output Table

| supplier_iban | purchase_order | supplier_email      | freight_amount | supplier_address             | receiver_address             | total_amount | supplier_name | total_tax_amount | payment_terms    | line_item                                                                     | receiver_name | receiver_email    | due_date   | invoice_date | invoice_id | currency | receiver_tax_id | net_amount | vat |
|---------------|----------------|---------------------|----------------|------------------------------|------------------------------|--------------|---------------|------------------|------------------|-------------------------------------------------------------------------------|---------------|-------------------|------------|--------------|------------|----------|-----------------|------------|-----|
|            50 |              1 | user@companyabc.com |            600 | 111 Main Street Anytown, USA | 222 Main Street Anytown, USA |         2140 | Company ABC   |              140 | 6 month contract | [Tool A 500 1.00 500.00,Service B 1 900.00 900.00,Resource C 50 12.00 600.00] | John Doe      | johndoe@email.com | 2025-01-01 |   1970-01-01 | NO. 001    | $        |               1 |       2000 | 140 |

### CODE REVIEW

File: google/cloud/documentai_toolbox/wrappers/document.py [hunk 1]
```diff
@@ -344,3 +345,64 @@ def get_entity_by_type(self, target_type: str) -> List[Entity]:
...
+        r"""Returns Dictionary of entities in document.
+
+        Returns:
+            Dict:
+                The Dict of the entities indexed by type.
+
+        """
+        entities_dict: Dict = {}
+        for entity in self.entities:
+            entity_type = entity.type_.replace("/", "_")
```
[the remark this answers is not in the document]
Comment (holtskinner) [reply google/cloud/documentai_toolbox/wrappers/document.py:359]: Nested Entities, which need to be added in to the extraction.

[the remark this answers is not in the document]
Comment (holtskinner) [reply google/cloud/documentai_toolbox/wrappers/document.py:382]: Added them to the entity type

File: google/cloud/documentai_toolbox/wrappers/document.py [hunk 2]
```diff
@@ -365,3 +368,66 @@ def get_entity_by_type(self, target_type: str) -> List[Entity]:
...
+        return entities_dict
+
+    def entities_to_bigquery(
+        self, dataset_name: str, table_name: str, project_id: Optional[str] = None
+    ) -> bigquery.job.LoadJob:
+        r"""Adds extracted entities to a BigQuery table.
+
+        Args:
+            dataset_name (str):
+                Name of the BigQuery dataset.
```
Comment (galz10) [inline google/cloud/documentai_toolbox/wrappers/document.py:406]: Nit: required tag to this

File: google/cloud/documentai_toolbox/wrappers/document.py [hunk 3]
```diff
@@ -365,3 +368,66 @@ def get_entity_by_type(self, target_type: str) -> List[Entity]:
...
+            table_name (str):
+                Name of the BigQuery table.
```
Comment (galz10) [inline google/cloud/documentai_toolbox/wrappers/document.py:408]: Nit: required tag to this

**Review by galz10 — APPROVED** [review]

**Pull request merged**